src/main/java/teammates/ui/controller/AdminAccountManagementPageAction.java

```diff
@@ -15,39 +15,40 @@
 
     @Override
     protected ActionResult execute() throws EntityDoesNotExistException {
-        
         new GateKeeper().verifyAdminPrivileges(account);
         
-        @SuppressWarnings("deprecation") //This method is deprecated to prevent unintended usage. This is an intended usage.
-        List<InstructorAttributes> allInstructorsList = logic.getAllInstructors();
-        @SuppressWarnings("deprecation") //This method is deprecated to prevent unintended usage. This is an intended usage.
-        List<AccountAttributes> allInstructorAccountsList = logic.getInstructorAccounts();
-        
-        boolean isToShowAll = this.getRequestParamAsBoolean("all");
+        String instructorGoogleId = this.getRequestParamValue("googleId");
         
         Map<String, ArrayList<InstructorAttributes>> instructorCoursesTable = new HashMap<String, ArrayList<InstructorAttributes>>();
         Map<String, AccountAttributes> instructorAccountsTable = new HashMap<String, AccountAttributes>();
         
-        for(AccountAttributes acc : allInstructorAccountsList){
-            instructorAccountsTable.put(acc.googleId, acc);
+        if (instructorGoogleId == null) {
```

Comment:
@bamboo3250, a minor point, but I guess we should also move this to right after instructorGoogleId is initialised